src/systemTest/java/com/flightstats/hub/testcase/WebhookLifecycleTest.java

```diff
@@ -0,0 +1,198 @@
+import com.flightstats.hub.client.ChannelItemResourceClient;
+import com.flightstats.hub.client.ChannelResourceClient;
+import com.flightstats.hub.client.WebhookResourceClient;
+import com.flightstats.hub.model.Channel;
+import com.flightstats.hub.model.ChannelItem;
+import com.flightstats.hub.model.Webhook;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import retrofit2.Call;
+import retrofit2.Response;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static javax.ws.rs.core.Response.Status.CREATED;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@Slf4j
+public class WebhookLifecycleTest extends BaseTest {
+
+    private static final String EMPTY_STRING = "";
+
+    private ChannelItemResourceClient channelItemResourceClient;
+    private ChannelResourceClient channelResourceClient;
+    private WebhookResourceClient webhookResourceClient;
+    private CallbackResourceClient callbackResourceClient;
+    private CallbackServer callbackServer;
+
+    private String channelName;
+    private String webhookName;
+
+    private List<String> channelItemsPosted;
+
+    @Before
+    public void setup() {
+
+        this.channelItemResourceClient = getHttpClient(ChannelItemResourceClient.class);
+        this.channelResourceClient = getHttpClient(ChannelResourceClient.class);
+        this.webhookResourceClient = getHttpClient(WebhookResourceClient.class);
+        this.callbackResourceClient = getHttpClient(CallbackResourceClient.class);
+        this.callbackServer = injector.getInstance(CallbackServer.class);
+
+        this.callbackServer.start();
+
+        this.channelName = generateRandomString();
+        this.webhookName = generateRandomString();
+    }
+
+    @Test
+    @SneakyThrows
+    public void testWebhookWithNoStartItem() {
+        final String data = "{\"webhook\": \"test1\", \"startitem\":\"false\"}";
+
+        createChannel();
+        addWebhook(buildWebhook(EMPTY_STRING));
+
+        List<String> channelItems = addItemsToChannel(data);
+        verifyWebhookCallback(channelItems);
+    }
+
+    @Test
+    @SneakyThrows
+    public void testWebhookWithStartItem() {
+        final String data = "{\"webhook\": \"test2\", \"startitem\":\"true\"}";
+
+        createChannel();
+        List<String> channelItems = addItemsToChannel(data);
+
+        addWebhook(buildWebhook(channelItems.get(4)));
+        verifyWebhookCallback(channelItems.subList(5, channelItems.size()));
+    }
+
+    @SneakyThrows
+    private void logWebhookCallbackError() {
+        log.info("Webhook name {} ", webhookName);
```

Comment:
Is this meant to test that errors are logged when a callback returns something other than a 2xx? If so, I'd probably create another callback endpoint that always returns a 4xx or 5xx and create a webhook with that as a callbackUrl That way when you add an item to the channel, you can confirm that the item uri shows up in the webhook errors list.  I think it'd mean you'll need to create a WebhookErrors model, though.

Reply:
this is not test for errors endpoint. when exception happens I am trying to log errors. After test chanel, webhook is all cleaned up and we do not have data to verify.

Reply:
Oh! I didn't read closely enough. Sorry about that.